### ISSUE TITLE
[bug] the rules az-property-type: gives false postive 

### ISSUE THREAD

**Issue by LasneF** [Azure/azure-api-style-guide#134]
given the rules az-property-type: 

this generate false positive as it catches properties in examples .

use case beeing that  i have an API with endpoints returning an object containing a field called 
"properties" as part of the payload 

for instance 

 
```
examples:
 Dog:
      description: |
        Creation of a Zero Coupon scenario 
      value:
        type: GOOD_DOG
        properties:
          size: TALL
          legs:
            - id : 1
```

returns          
  warning  az-property-type       Property should have a defined type.             components.examples.Dog.value.properties.legs
